src/kernels/MIOpenTensorKernels.cl

```diff
@@ -395,102 +272,46 @@ __kernel void OpTensorLeadingOnesGeneric(global MIOPEN_TYPE* a,
     global MIOPEN_TYPE* a_off = a + Aoffset;
     global MIOPEN_TYPE* b_off = b + Boffset;
     global MIOPEN_TYPE* c_off = c + Coffset;
-#if FIRST_NOT_ONE == 3 // bitmap = 1,1,1,1
-    int tid = get_global_id(0);
-    // MIOPEN_TYPE operand = b[tid];
 
-    // num_wg: the number of workgroups should be launched
-    // MAX_NUM_WG: the maximum number of workgroups actually launched
-    for(; tid < num_wg; tid += MAX_NUM_WG)
-    {
-
-        int o_w = tid % c_w;
-        int o_h = (tid / c_w) % c_h;
-        int o_c = (tid / (c_w * c_h)) % c_c;
-        int o_n = tid / (c_w * c_h * c_c);
-
-        int aindex = o_n * a_nstride + o_c * a_cstride + o_h * a_hstride + o_w;
-        int bindex = o_n * b_nstride + o_c * b_cstride + o_h * b_hstride + o_w;
-        int cindex = o_n * c_nstride + o_c * c_cstride + o_h * c_hstride + o_w;
-        c_off[cindex] =
-            MIOPEN_TENSOR_OP(alpha0 * a_off[aindex], alpha1 * b_off[bindex]) + beta * c_off[cindex];
-    }
-
-#elif FIRST_NOT_ONE == 2 // bitmap = 1,1,1,0
-    int gid = get_group_id(0);
+    int gid = (bitmap == 15) ? get_global_id(0) : get_group_id(0);
 
     // num_wg: the number of workgroups should be launched
     // MAX_NUM_WG: the maximum number of workgroups actually launched
     for(; gid < num_wg; gid += MAX_NUM_WG)
     {
-        int lid = get_local_id(0);
-        int o_h = gid % c_h;
-        int o_c = (gid / c_h) % c_c;
-        int o_n = gid / (c_c * c_h);
-
-        int bindex          = o_n * b_nstride + o_c * b_cstride + o_h * b_hstride;
-        MIOPEN_TYPE operand = b_off[bindex] * alpha1;
-
-        while(lid < work_per_wg)
-        {
-            int aindex = o_n * a_nstride + o_c * a_cstride + o_h * a_hstride + lid;
-            int cindex = o_n * c_nstride + o_c * c_cstride + o_h * c_hstride + lid;
-            c_off[cindex] =
-                MIOPEN_TENSOR_OP(alpha0 * a_off[aindex], operand) + beta * c_off[cindex];
-
-            lid += get_local_size(0);
-        }
-    }
-#elif FIRST_NOT_ONE == 1 // bitmap = 1,1,0,0
-    int gid = get_group_id(0);
-
-    // num_wg: the number of workgroups should be launched
-    // MAX_NUM_WG: the maximum number of workgroups actually launched
-    for(; gid < num_wg; gid += MAX_NUM_WG)
-    {
-        int lid = get_local_id(0);
-        int o_c = gid % c_c;
-        int o_n = gid / c_c;
-
-        int bindex          = o_n * b_nstride + o_c * b_cstride;
+        int lid    = (bitmap == 15) ? 0 : get_local_id(0);
+        int lcl_sz = (bitmap == 15) ? work_per_wg : get_local_size(0);
+
+        int o_w = (bitmap & (1 << 0)) ? (gid % c_w) : 0;
```

Comment:
You could replace 1 << 0  with just 0x1 - in case the compiler fails to optimize it.
You could replace 1 << 1  with just 0x2 - in case the compiler fails to optimize it. 

Personally, I find 0x1/0x2 easier to read than 1 << 0/ 1 << 1

Reply:
I would prefer keep the current format as an indicator that this is a bitmap operation.